module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,62 @@
+#include "compression.hpp"
+
+#include <algorithm>
+#include <iostream>
+#include <utility>
+
+using vector_of_pairs = std::vector<std::pair<uint8_t, uint8_t>>;
+using array_of_arrays = std::array<std::array<uint8_t, width>, height>;
+
+vector_of_pairs compressGrayscale(const array_of_arrays& bitmap) {
+    vector_of_pairs compressed{};
+    compressed.reserve(width * height);
+    uint8_t occurrences = 0;
+    uint8_t color_code = 0;
+    for (const auto& row : bitmap) {
+        color_code = row.front();
+        occurrences = 1;
+        for (auto element = row.begin(); element != row.end(); ++element) {
+            auto next = std::next(element);
+            if (color_code != *next || next == row.end()) {
```

Comment:
if next is row.end() what will be the value of *next ? consider the value being color_node.

Reply:
@nauka-programowania-MA coulb you please comment on that

Reply:
@ziobron Can you have a look?